cassandra/cqltypes.py

```diff
@@ -625,6 +633,95 @@ def serialize(timeuuid, protocol_version):
             raise TypeError("Got a non-UUID object for a UUID value")
 
 
+class SimpleDateType(_CassandraType):
+    typename = 'date'
+    seconds_per_day = 60 * 60 * 24
+    date_format = "%Y-%m-%d"
+
+    @classmethod
+    def validate(cls, val):
+        if isinstance(val, six.string_types):
+            val = cls.interpret_simpledate_string(val)
+        elif (not isinstance(val, datetime.date)) and (type(val) not in _date_types):
+            raise TypeError('SimpleDateType arg must be a datetime.date, unsigned integer, or string in the format YYYY-MM-DD')
+        return val
+
+    @staticmethod
+    def interpret_simpledate_string(v):
+        date_time = datetime.datetime.strptime(v, SimpleDateType.date_format)
+        return datetime.date(date_time.year, date_time.month, date_time.day)
+
+    @staticmethod
+    def serialize(val, protocol_version):
+        # Values of the 'date'` type are encoded as 32-bit unsigned integers
+        # representing a number of days with "the epoch" at the center of the
+        # range (2^31). Epoch is January 1st, 1970
+        try:
+            shifted = (calendar.timegm(val.timetuple()) // SimpleDateType.seconds_per_day) + 2 ** 31
```

Comment:
Docs should note that datetimes that don't correspond exactly to a date are truncated (as opposed to being rounded).  We don't have great docs around the handling of various types yet, though.